File: ChatTemplate/Models/Battlefield.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ChatTemplate.Models
{
    public class Battlefield
    {
        public ICollection<ICollection<Cell>> Cells { get; set; }
        public ICollection<Ship> Ships { get; set; }
    }
}

File: ChatTemplate/ClientApp/src/app/models/battlefield.ts
import { CellStatus } from './cell-status.enum';
import { Cell } from './cell';
import { Ship } from './ship';

export class Battlefield {

	public cells: Cell[][] = [];
	public ships: Ship[] = [];

	constructor(fieldSize: number) {
        for(let i: number = 0; i < fieldSize; i++) {
            this.cells[i] = [];
            for(let j: number = 0; j < fieldSize; j++) {
                this.cells[i][j] = new Cell(j, i);
                this.cells[i][j].status = CellStatus.clear;
            }
        }
	}
}
